dot/sync/fullsync.go

```diff
@@ -395,85 +407,79 @@ type RequestResponseData struct {
 	responseData []*types.BlockData
 }
 
-func validateResults(results []*SyncTaskResult, badBlocks []string) (repChanges []Change,
-	peersToBlock []peer.ID, validRes []RequestResponseData) {
-
-	repChanges = make([]Change, 0)
-	peersToBlock = make([]peer.ID, 0)
-	validRes = make([]RequestResponseData, 0, len(results))
-
-resultLoop:
-	for _, result := range results {
-		request := result.request.(*messages.BlockRequestMessage)
+func validateResult(result TaskResult, badBlocks []string) (repChange *Change,
+	blockPeer bool, validRes *RequestResponseData) {
 
-		if !result.completed {
-			continue
-		}
-
-		response := result.response.(*messages.BlockResponseMessage)
-		if request.Direction == messages.Descending {
-			// reverse blocks before pre-validating and placing in ready queue
-			slices.Reverse(response.BlockData)
-		}
+	if !result.Completed {
+		return
+	}
 
-		err := validateResponseFields(request, response.BlockData)
-		if err != nil {
-			logger.Warnf("validating fields: %s", err)
-			// TODO: check the reputation change for nil body in response
-			// and nil justification in response
-			if errors.Is(err, errNilHeaderInResponse) {
-				repChanges = append(repChanges, Change{
-					who: result.who,
-					rep: peerset.ReputationChange{
-						Value:  peerset.IncompleteHeaderValue,
-						Reason: peerset.IncompleteHeaderReason,
-					},
-				})
-			}
+	task, ok := result.Task.(*syncTask)
+	if !ok {
+		logger.Warnf("skipping unexpected task type in TaskResult: %T", result.Task)
+		return
+	}
 
-			continue
-		}
+	request := task.request.(*messages.BlockRequestMessage)
```

Comment:
you can skip this cast if you change the `syncTask.request` from `messages.P2PMessage` to `*messages.BlockRequestMessage`